### ISSUE TITLE
chronos_test doesn't use the locally-built libcares

### ISSUE THREAD

**Issue by rkd-msw** [Metaswitch/chronos#209]
```
clang++-3.6  -o build/bin/chronos_test build/chronos_test/chronos_internal_connection.o build/chronos_test/handlers.o build/chronos_test/replicator.o build/chronos_test/timer_handler.o build/chronos_test/globals.o build/chronos_test/http_callback.o build/chronos_test/timer.o build/chronos_test/timer_store.o build/chronos_test/log.o build/chronos_test/logger.o build/chronos_test/unique.o build/chronos_test/signalhandler.o build/chronos_test/alarm.o build/chronos_test/httpstack.o build/chronos_test/httpstack_utils.o build/chronos_test/accesslogger.o build/chronos_test/utils.o build/chronos_test/health_checker.o build/chronos_test/exception_handler.o build/chronos_test/httpconnection.o build/chronos_test/statistic.o build/chronos_test/baseresolver.o build/chronos_test/dnscachedresolver.o build/chronos_test/dnsparser.o build/chronos_test/zmq_lvc.o build/chronos_test/httpresolver.o build/chronos_test/counter.o build/chronos_test/snmp_scalar.o build/chronos_test/snmp_agent.o build/chronos_test/snmp_row.o build/chronos_test/timer_counter.o build/chronos_test/MurmurHash3.o build/chronos_test/base.o build/chronos_test/fakesnmp.o build/chronos_test/test_globals.o build/chronos_test/test_handler.o build/chronos_test/test_main.o build/chronos_test/test_timer.o build/chronos_test/test_timer_handler.o build/chronos_test/test_timer_replica_choosing.o build/chronos_test/test_timer_store.o build/chronos_test/timer_helper.o build/chronos_test/test_interposer.o build/chronos_test/fakelogger.o build/chronos_test/mock_sas.o build/chronos_test/fakecurl.o build/chronos_test/pthread_cond_var_helper.o build/chronos_test/mock_increment_table.o build/chronos_test/mock_infinite_table.o build/chronos_test/mock_scalar_table.o build/obj/gmock-all.o build/obj/gtest-all.o -lgcov --coverage -Lbuild/usr/lib -lrt -lpthread -lcurl -levent -lboost_program_options -lboost_regex -lzmq -lc -lboost_filesystem -lboost_system -levhtp -levent_pthreads -lcares -L/usr/lib/x86_64-linux-gnu -lnetsnmpmibs -lnetsnmpagent -lnetsnmp -ldl
LD_LIBRARY_PATH=../usr/lib/ valgrind --gen-suppressions=all --leak-check=full --track-origins=yes --malloc-fill=cc --free-fill=df --xml=yes --xml-file=build/chronos_test/valgrind_output.xml build/bin/chronos_test --gtest_filter="-*DeathTest*"
build/bin/chronos_test: error while loading shared libraries: libcares.so.2: cannot open shared object file: No such file or directory
```

Looks like `sudo apt-get install libc-ares2` is a workaround.

**Comment by bossmc** [comment]
But not a good workaround... We build a custom copy of `c-ares` but we don't place it in `../usr/lib` as we specify in `LD_LIBRARY_PATH`, instead it's in `build/usr/lib/`.  Need to make the LD_LIBRARY_PATH customizable in `build-infra/cpp.mk`.

**Comment by bossmc** [comment]
Fix by #212 (with the associated build-infra change).
